ocaml/xapi/xapi_vgpu.ml

```diff
@@ -21,12 +21,12 @@ let m = Mutex.create ()
 let valid_device device =
   device = "0"
 
-let create ~__context  ~vM ~gPU_group ~device ~other_config ~_type =
+let create' ~__context  ~vM ~gPU_group ~device ~other_config ~_type ~expected_vm_state =
```

Comment:
Why not
 
```ocaml
let create ~__context  ~vM ~gPU_group ~device ~other_config ~_type ?(expected_vm_state = `Halted)
```
instead?

Reply:
This had caused an issue in the automatically-generated code in api_server.ml:
```
(** Actions module *)
module Actions = struct
  ...
  module VGPU = Xapi_vgpu
end
```
that fails to compile with
```
File "ocaml/xapi/api_server.ml", line 100, characters 47-54:
Error: Signature mismatch:
       ...
       In module VGPU:
       Values do not match:
         val create :
           __context:Context.t ->
           vM:[ `VM ] Ref.t ->
           gPU_group:[ `GPU_group ] Ref.t ->
           device:string ->
           other_config:(string * string) list ->
           _type:[ `VGPU_type ] Ref.t ->
           ?expected_vm_state:[< `Halted
                               | `Migrating
                               | `Paused
                               | `Running
                               | `ShuttingDown
                               | `Suspended
                               > `Halted `Paused `Running `Suspended ] ->
           unit -> [ `VGPU ] Ref.t
       is not included in
         val create :
           __context:Context.t ->
           vM:API.ref_VM ->
           gPU_group:API.ref_GPU_group ->
           device:string ->
           other_config:API.string_to_string_map ->
           _type:API.ref_VGPU_type -> API.ref_VGPU
```
the vgpu object in the datamodel is defined with
```
let vgpu =
  let create = call
      ~name:"create"
      ~versioned_params:[
        {param_type=(Ref _vm); param_name="VM"; param_doc=""; param_release=boston_release; param_default=None};
        {param_type=(Ref _gpu_group); param_name="GPU_group"; param_doc=""; param_release=boston_release; param_default=None};
        {param_type=String; param_name="device"; param_doc=""; param_release=boston_release; param_default=Some (VString "0")};
        {param_type=(Map (String, String)); param_name="other_config"; param_doc=""; param_release=boston_release; param_default=Some (VMap [])};
        {param_type=(Ref _vgpu_type); param_name="type"; param_doc=""; param_release=vgpu_tech_preview_release; param_default=(Some (VRef null_ref))};
      ]
```
I thought that would be better not to change the datamodel to match a new signature for `xapi_vgpu.create()`. -- instead, I kept the signature of `xapi_vgpu.create()` and used an internal function instead to handle the extra parameter.

Reply:
Argh, yes if it has to play with the autogenerated code, let's go with the initial version. Please add also a comment on `create'` that explains why we need to duplicate it in such way.

Reply:
done: df49140bfdc1f12e8b344e8f1be3fa00c58f8e33